emitterc.go

```diff
@@ -361,6 +361,9 @@ func (emitter *yamlEmitter) emitStreamStart(event *yamlEvent) bool {
 // Expect DOCUMENT-START or STREAM-END.
 func (emitter *yamlEmitter) emitDocumentStart(event *yamlEvent, first bool) bool {
 	if event.typ == yaml_DOCUMENT_START_EVENT {
+		// The next element has been parsed because yaml_emitter_need_more_events() demands
+		// one more element for yaml_DOCUMENT_START_EVENT (accumulate == 1)
+		isEmpty := emitter.events[emitter.events_head + 1].typ == yaml_DOCUMENT_END_EVENT
```

Comment:
I saw the comment, but isn't safer to consider something like this ?

```suggestion
		isEmpty := len(emitter.events) < emitter.events_head + 1 || emitter.events[emitter.events_head + 1].typ == yaml_DOCUMENT_END_EVENT
```

I'm not sure about the +1 and <, maybe it's +2, or <=
But I think you will get the point

Reply:
Yes, but we don't know how people can use the lib. So I appreciate you added the check on the length.